### ISSUE TITLE
Improve rule S5659 (`insecure-jwt-token`): should support "jose" 

### ISSUE THREAD

**Issue by agigleux** [SonarSource/SonarJS#2252]
S5659 should detect the use of the JWK.None / empty algorithm on the following functions:

[JWT.sign(payload, key[, options])](https://github.com/panva/jose/blob/HEAD/docs/README.md#jwtsignpayload-key-options)
* `algorithm` property should be provided with a strong cipher

[JWT.verify(token, keyOrStore[, options])](https://github.com/panva/jose/blob/HEAD/docs/README.md#jwtverifytoken-keyorstore-options)
* `algorithms` property should be provided with a strong cipher

Noncompliant Example

```
const jose = require("jose");
const { JWK, JWT } = jose;
const token = JWT.verify('token', JWK.None);
```

Compliant Example

* https://www.npmjs.com/package/jose#jwt-signing
* https://github.com/panva/jose#jwt-verifying

**Comment by yassin-kammoun-sonarsource** [comment]
@agigleux The API has significantly changed since the ticket was created. Both `JWT.verify` and `JWT.sign` no longer exist. As a result, the non-compliant example is irrelevant now. Does it still make sense to keep this ticket ?

**Comment by vilchik-elena** [comment]
@agigleux could you confirm if this ticket makes sense?

**Comment by ilia-kebets-sonarsource** [comment]
As mentioned here, we can close this: https://github.com/SonarSource/SonarJS/issues/2252#issuecomment-822409003

**Comment by yassin-kammoun-sonarsource** [comment]
Yes, let's close it.
